File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

// Prototypes
void space(int n);
void hex(int n);

int main(void)
{
    // Tnitialize value
    int height = 0;

    // Require user input
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Generate lines of space and hex
    for (int i = 1; i <= height; i++)
    {
        space(height - i);
        hex(i);
        space(2);
        hex(i);
        printf("\n");
    }
}

// Generate space based on integer input
void space(int n)
{
    while (n > 0)
    {
        printf(" ");
        n--;
    }
}

// Generate hex based on integer input
void hex(int n)
{
    while (n > 0)
    {
        printf("#");
        n--;
    }
}

File: pset6/mario/mario.py
from cs50 import get_int


def main():

    # Get user input for height
    while True:
        h = get_int("Height: ")
        if 1 <= h <= 8:
            break

    # Print lines
    for i in range(h):
        space(h - i - 1)
        sign(i + 1)
        space(2)
        sign(i + 1)
        print()

# Print spaces


def space(n):
    for _ in range(n):
        print(" ", end="")

    return

# Print spaces


def sign(n):
    for _ in range(n):
        print("#", end="")

    return


if __name__ == "__main__":
    main()